lib/request.js

```diff
@@ -209,23 +212,28 @@ Object.defineProperties(Request.prototype, {
   },
   host: {
     get () {
-      return this.raw.headers.host || this.raw.headers[':authority']
+      let host = this.headers.host ?? this.headers[':authority']
+      // support http.requireHostHeader === false
+      if (this.server.server.requireHostHeader === false) host ??= ''
+      return host
     }
   },
   hostname: {
     get () {
-      return (this.host).split(':')[0]
+      return this.host.split(':')[0]
     }
   },
   port: {
     get () {
       // first try taking port from host
-      const portFromHost = parseInt((this.host).split(':').slice(-1)[0])
+      const portFromHost = parseInt(this.host.split(':').slice(-1)[0])
       if (!isNaN(portFromHost)) {
         return portFromHost
       }
       // now fall back to port from host/:authority header
-      const portFromHeader = parseInt((this.headers.host || this.headers[':authority']).split(':').slice(-1)[0])
+      let host = (this.headers.host ?? this.headers[':authority'])
```

Comment:
can we avoid it?
```suggestion
      const host = this.host
```

Reply:
They actually not the same, `this.headers` can have additional headers added by users.
But `host` and `hostname` always  respect the original headers.
I don't knows why it makes this way, but I will keep it for now.

Reply:
It's about forwarded headers, if they fail it tries the originals